lib/wasi/src/runtime/mod.rs

```diff
@@ -97,13 +123,21 @@ impl PluggableRuntimeImplementation {
                 let http_client = None;
             }
         }
+        cfg_if::cfg_if! {
```

Comment:
I went through quite a bit of work to remove `#[cfg()]` code like this from this generic type. 

Let's please keep this out of here.
Construction code like the in the CLI can set this.

Reply:
Fair enough - I'll look into how to get rid of this

Reply:
removed